lib/pf/pfcmd/checkup.pm

```diff
@@ -1101,8 +1103,9 @@ sub portal_profiles {
                 if ( $key !~ /$profile_params/ );
             if ($key eq 'filter') {
                 foreach my $filter (@{$data->{filter}}) {
-                    add_problem( $FATAL, "Filter '$filter' is invalid for profile '$portal_profile' please update to newer format 'type:data'" )
-                        unless $filter =~ $pf::factory::condition::profile::PROFILE_FILTER_REGEX;
+                    my ($rc, $message) = $validator->validate($filter);
+                    add_problem( $FATAL, "Filter '$filter' is invalid for profile '$portal_profile': $message ")
+                        unless $rc;
```

Comment:
Not sure it is really better to put it on a new line below. Can be easy to miss to the untrained eye